aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -441,6 +444,150 @@ std::tuple<Tensor&,Tensor&> min_out(Tensor& min, Tensor& min_indices,
   return result;
 }
 
+static Tensor scalar_tensor_like(const Tensor& like, const Scalar& value) {
+  // FIXME: This should do type promotion with value, but that would be a BC-break
+  return at::scalar_tensor(value, at::device(like.device()).dtype(like.scalar_type()));
+}
+
+static c10::optional<Tensor> scalar_tensor_like(const Tensor& like, const c10::optional<Scalar>& value) {
+  if (value) {
+    return scalar_tensor_like(like, *value);
+  }
+  return c10::nullopt;
+}
+
+Tensor& clamp_out(const Tensor& self, c10::optional<Scalar> min, c10::optional<Scalar> max, Tensor& result) {
+  return at::clamp_outf(self, scalar_tensor_like(self, min), scalar_tensor_like(self, max), result);
```

Comment:
Can the CUDA clamp kernel not handle scalar inputs?

Reply:
IIUC `gpu_kernel` requires all inputs to be on CUDA but `gpu_kernel_with_scalars` allows CPU-scalars. However, that only works for binary kernels:
https://github.com/pytorch/pytorch/blob/0569f638fe613ec4ef8294f0743223726958aa4e/aten/src/ATen/native/cuda/Loops.cuh#L144-L145

Reply:
@mruberry I can add scalar support all the same though, if it's desirable. Bear in mind it does mean each kernel has to be duplicated for all combinations of CUDA-tensor and CPU-scalar.

Reply:
Yeah that sounds like a pain.

I wonder if we should limit ourselves to just the scalar x scalar and tensor x tensor cases for simplicity?